src/taipy/gui/renderers/builder.py

```diff
@@ -60,21 +60,25 @@ class _Builder:
 
     __BLOCK_CONTROLS = ["dialog", "expandable", "pane", "part"]
 
-    __CHART_AXIS_SUBSTITUTION = {
-        "scattermapbox": {"x": "lon", "y": "lat"},
-        "scattergeo": {"x": "lon", "y": "lat"},
-        "densitymapbox": {"x": "lon", "y": "lat"},
-        "scatterpolar": {"x": "r", "y": "theta"},
-        "scatterpolargl": {"x": "r", "y": "theta"},
+    __CHART_AXIS_SUBSTITUTION: t.Dict[str, t.List[str]] = {
+        "scattermapbox": ["lon", "lat"],
+        "scattergeo": ["lon", "lat"],
+        "densitymapbox": ["lon", "lat"],
+        "scatterpolar": ["r", "theta"],
+        "scatterpolargl": ["r", "theta"],
+        "candlestick": ["x", "close", "open", "high", "low"],
     }
-    __CHART_AXIS_SUBSTITUTION_INDEXES = {
-        "scattermapbox": ((0, 18), (1, 19)),
-        "scattergeo": ((0, 18), (1, 19)),
-        "densitymapbox": ((0, 18), (1, 19)),
-        "scatterpolar": ((0, 21), (1, 22)),
-        "scatterpolargl": ((0, 21), (1, 22)),
+    __CHART_AXIS_SUBSTITUTION_INDEXES: t.Dict[str, t.Iterable[int]] = {
```

Comment:
Don't you think now would be a good time to have symbols to hold those indices?

Reply:
I don't see a real value in this ...

Reply:
OK switching to an enum